### DISCUSSION TITLE
Add instrument icons for vibraphone and marimba

### DISCUSSION THREAD

**Discussion by vibraphon** [jamulussoftware/jamulus#2043] [Ideas]
I am missing instrument icons for vibraphone and marimba.
May be it is possible to add them.

**Comment by ghost** [comment]
Hi, I remember a similar request a while ago. I looked at various vibraphones and marimbas to see if I could scale an image down. Maybe someone who is good at low-res art could make an image... Maybe of a few 'tone plates' or 'piano key shapes' with crossed mallets.

**Reply by gilgongo** [reply]
If anyone comes up with one feel free to submit it.

**Comment by vibraphon** [comment]
Here is a small picture of my homemade vibraphone. This is definitely free of copyright claims. But I don't know if it is suitable for Jamulus.
![Jamulus_05](https://user-images.githubusercontent.com/91273194/134854989-e489cda2-d3c4-4c6f-9a7c-0ed29583d7e4.jpg)

**Reply by ghost** [reply]
I hope it works!

**Reply by gilgongo** [reply]
Thanks @vibraphon - do you have a hi-res version? Might make for better quality when it gets shrunk.

**Reply by ghost** [reply]
I think the vibraphone only needs several pixels manually touched up. I also think it is shrunk to its minimum, but I'm not a graphic artist.
Edit: The picture should be tested on Jamulus right now because it may be acceptable.

**Reply by vibraphon** [reply]
Of course. I had specially tried to reduce the size of the image.
![Jamulus_03](https://user-images.githubusercontent.com/91273194/134943284-73da07c4-2cce-4d9e-b27a-4b1732c0fd7a.jpg)

**Reply by vibraphon** [reply]
Do you want me to test it?
I don't see how I can test that.

**Reply by gilgongo** [reply]
Would it be possble to remove the shadow? It gets in the way a bit at 35px (not that 35px a standard width, but some of them are): 
![image](https://user-images.githubusercontent.com/4561747/134946947-af8d92ed-71e4-4b2b-98d7-21d4f1c536db.png)

**Reply by gilgongo** [reply]
If you want to test it, you'll need to do that in your own fork and compile it. See [this example](https://github.com/jamulussoftware/jamulus/commit/30acb6a93030a61e6f9df8d3141f5cdbd9b17427) from a pevious addition.

**Reply by vibraphon** [reply]
Oh, I don't know anything about programming at all. I have tried to take out the shadow. Here are 3 different image sizes of the same file.
I hope you can do something with it.

![Jamulus_03a](https://user-images.githubusercontent.com/91273194/134949750-e57f19d5-800c-45d9-8842-473178ac9416.jpg)
![Jamulus_03a_50](https://user-images.githubusercontent.com/91273194/134949788-27cf0aef-4920-4249-9e24-2fb979580779.jpg)
![Jamulus_03a_35](https://user-images.githubusercontent.com/91273194/134949804-a88f0ead-b259-457e-ad4d-eddfbd7e1c12.jpg)

**Reply by ghost** [reply]
![vibraphone](https://user-images.githubusercontent.com/73317641/134960267-49e5c3ee-ed7d-4c1a-ae43-278f5269aec2.png)
**Tested OK** (Jamulus - My Profie, and Skin:Compact,Fancy,Normal)
Rename the above picture to vibraphone.png (I converted from .jpg to .png)
Follow gilgongo's procedure.
YES !!!
**Edit:**
I like the white background for the vibraphone (please don't change it to to transparent as contrast is lost).

**Furthermore,** the none.png picture with the (question mark - ?) looks like its transparency has been polluted with another picture.
**Edit:**
![none](https://user-images.githubusercontent.com/73317641/135001712-0495a9f0-b3c0-4371-8452-81eb0d2140de.png)
**none.prg fixed** (rename above picture to none.png and put in /src/rec/instruments/none.png)

**Found Warning while compiling:**
src/socket.cpp: In member function ‘void CSocket::SendPacket(const CVector<unsigned char>&, const CHostAddress&)’:
src/socket.cpp:325:13: warning: ‘status’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             if ( status >= 0 )
             ^

**Reply by vibraphon** [reply]
Thank you for testing, but I absolutely don't know what I have to do.
Unfortunately I am 60+ and don't understand the procedure, sorry.

I just hope that the icon will show up in the selection list at some point!

**Reply by vibraphon** [reply]
Sorry for the stupid question:
But what can or must I do to make the icon generally available in Jamulus at some point?
Unfortunately I have no idea about programming and compiling.

**Reply by ghost** [reply]
To test Jamulus' vibraphone version on Windows:
- sign in to Jamulus like you would to reply to this message.
- go to the link below and download the ASIO jamulus_win.exe (or JACK for more testing jamulus_win_jack.exe)
https://github.com/jamulussoftware/jamulus/actions/runs/1374306785

**Comment by ghost** [comment]
**Jamulus will have a Vibraphone**
Add Vibraphone icon to instruments. See PR #2080

**Reply by vibraphon** [reply]
Thank you very much!
It's great!

**Comment by vibraphon** [comment]
Unfortunately, I only see the vibraphone icon on my computer. My partners do not see it even after the current update.
What do I have to do to make the icon visible and usable?

**Reply by gilgongo** [reply]
Unfortuntately, the PR didn't make it into the last release. It's been merged for the next one though (3.9.0)
